cirq-google/cirq_google/engine/abstract_local_processor.py

```diff
@@ -0,0 +1,410 @@
+        """Retrieve a reservation given its id."""
+        if reservation_id in self._reservations:
+            return self._reservations[reservation_id]
+        else:
+            return None
+
+    def update_reservation(
+        self,
+        reservation_id: str,
+        start_time: datetime.datetime = None,
+        end_time: datetime.datetime = None,
+        whitelisted_users: List[str] = None,
+    ):
+        """Updates a reservation with new information.
+
+        Updates a reservation with a new start date, end date, or
+        list of additional users.  For each field, it the argument is left as
+        None, it will not be updated.
+        """
+        if reservation_id not in self._reservations:
+            raise ValueError(f'Reservation id {reservation_id} does not exist.')
+        if start_time:
+            self._reservations[reservation_id].start_time.seconds = _to_timestamp(start_time)
+        if end_time:
+            self._reservations[reservation_id].end_time.seconds = _to_timestamp(end_time)
+        if whitelisted_users:
+            del self._reservations[reservation_id].whitelisted_users[:]
+            self._reservations[reservation_id].whitelisted_users.extend(whitelisted_users)
+
+    def list_reservations(
+        self,
+        from_time: Union[None, datetime.datetime, datetime.timedelta] = datetime.timedelta(),
+        to_time: Union[None, datetime.datetime, datetime.timedelta] = datetime.timedelta(weeks=2),
+    ):
+        """Retrieves the reservations from a processor.
+
+        Only reservations from this processor and project will be
+        returned. The schedule may be filtered by starting and ending time.
+
+        Args:
+            from_time: Filters the returned reservations to only include entries
+                that end no earlier than the given value. Specified either as an
+                absolute time (datetime.datetime) or as a time relative to now
+                (datetime.timedelta). Defaults to now (a relative time of 0).
+                Set to None to omit this filter.
+            to_time: Filters the returned reservations to only include entries
+                that start no later than the given value. Specified either as an
+                absolute time (datetime.datetime) or as a time relative to now
+                (datetime.timedelta). Defaults to two weeks from now (a relative
+                time of two weeks). Set to None to omit this filter.
+
+        Returns:
+            A list of reservations.
+        """
+        start_timestamp = _to_timestamp(from_time)
+        end_timestamp = _to_timestamp(to_time)
+        reservation_list = []
+        for reservation in self._reservations.values():
+            if start_timestamp and reservation.start_time.seconds < start_timestamp:
+                continue
+            if end_timestamp and reservation.end_time.seconds > end_timestamp:
+                continue
```

Comment:
These are flipped - `reservation` should be excluded if its `end_time` is before `start_timestamp` or if its `start_time` is after `end_timestamp`.